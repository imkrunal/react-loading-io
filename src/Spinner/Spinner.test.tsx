import React from 'react'
import renderer from 'react-test-renderer'
import { Spinner } from './'

it('renders correctly', (): void => {
  const tree = renderer.create(<Spinner />).toJSON()
  expect(tree).toMatchSnapshot()
})

it('joins default class with className prop', (): void => {
  const { root } = renderer.create(<Spinner className='spinner' />)

  expect(root.findByType('div').props.className).toEqual('rl-spinner spinner')
})

it('sets CSS vars with props', (): void => {
  const size = 100

  const { root } = renderer.create(
    <Spinner size={size} color='#d36ac2' speed={2} style={{ margin: 20 }} />
  )
  const { style } = root.findByType('div').props

  const scale = size / 200
  const translate = size * scale

  expect(style).toEqual({
    '--rl-spinner-size': '100px',
    '--rl-spinner-color': '#d36ac2',
    '--rl-spinner-scale': scale,
    '--rl-spinner-translate': `${translate}px`,
    '--rl-spinner-speed': `2s`,
    margin: 20
  })
})
