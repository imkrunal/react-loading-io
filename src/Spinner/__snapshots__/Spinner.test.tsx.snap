// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`renders correctly 1`] = `
<div
  className="rl-spinner"
  style={
    Object {
      "--rl-spinner-color": "#ED4585",
      "--rl-spinner-scale": 1,
      "--rl-spinner-size": "200px",
      "--rl-spinner-speed": "1s",
      "--rl-spinner-translate": "200px",
    }
  }
>
  <div />
  <div />
  <div />
  <div />
  <div />
  <div />
  <div />
  <div />
  <div />
  <div />
  <div />
  <div />
</div>
`;
